components/ai/code/ai-bbe/natural_programming_vs_code_features.md

```diff
@@ -0,0 +1,16 @@
+---
+title: "Natural Programming: Bridge Requirements and Code"
+description: "Powered by the integration of Generative AI into the programming system, natural programming bridges requirements, documentation, and code to maintain continuous alignment between implementation and requirements.
+
+
+- **AI-Powered Code Generation** – Generate code against high-level business user requirements.
+
+
+- **AI-Powered Test Generation** – Generate Ballerina tests against business user requirements, enabling identification of implementation deviation from the requirements.
+
+
+- **Drift Detection & Fixes** – Proactively identify drift between requirements, code, and documentation. Use smart fix suggestions via VS Code to review and accept AI-powered suggestions to address drift.  
+
+
+- **Intent Preservation** – Maintain an effective summary of the prompts used to generate code, providing insight into developer input."
+---
```

Comment:
This is too wordy, and the rule of thumb we are following for blade text is to maintain less than 200 characters. We are deviating now, but can't do this longer. Can we consider rewording? One suggestion is below. 

```
Natural programming with AI

Leverage Generative AI to align code, requirements, and documentation.

AI-powered code generation – Generate code from high-level business requirements.

AI-powered test generation – Create Ballerina tests from business requirements to detect deviations.

Drift detection and fixes – Identify and fix mismatches between code, requirements, and docs using AI suggestions in VS Code.

Intent preservation – Keep a summary of prompts used for code generation to track developer intent.
```

Reply:
Minor edits

```
Natural programming with AI

Leverage Generative AI to keep requirements, code, and documentation in sync.

AI-powered code generation – Generate code from high-level business requirements.

Drift detection and fixes – Proactively identify drift between requirements, code, and documentation and use AI-powered suggestion on VS Code to address drift.

AI-powered test generation – Generate Ballerina tests against business requirements to enable test-driven drift detection.

Intent preservation – Maintain an effective summary of the prompts used for code generation to preserve user intention.
```

Reply:
+1 for content, but i think the previous topic is more attractive than the just `Natural programming with AI`
wdyt?

Reply:
+1 since natural programming is a new concept also.

Reply:
@SasinduDilshara can you switch drift and test generation as suggested in my comment? Otherwise "test-driven drift detection" might not make sense.

Reply:
Updated